src/python/pants/backend/python/tasks/pytest_run.py

```diff
@@ -378,34 +378,33 @@ def _pants_pytest_plugin_args(self, sources_map):
     # uses all arguments that look like paths to compute its rootdir, and we want
     # it to pick the buildroot.
     with temporary_dir(root_dir=self.workdir) as comm_dir:
-      rootdir_comm_path = os.path.join(comm_dir, 'pytest_rootdir.path')
-
-      def get_pytest_rootdir():
-        with open(rootdir_comm_path, 'r') as fp:
-          return fp.read()
-
       sources_map_path = os.path.join(comm_dir, 'sources_map.json')
       mode = 'w' if PY3 else 'wb'
       with open(sources_map_path, mode) as fp:
         json.dump(sources_map, fp)
 
       renaming_args = [
-        '--pants-rootdir-comm-path', rootdir_comm_path,
         '--pants-sources-map-path', sources_map_path
       ]
 
-      yield renaming_args + self._get_sharding_args(), get_pytest_rootdir
+      yield renaming_args + self._get_sharding_args()
 
   @contextmanager
   def _test_runner(self, workdirs, test_targets, sources_map):
     pytest_binary = self.context.products.get_data(PytestPrep.PytestBinary)
-    with self._pants_pytest_plugin_args(sources_map) as (plugin_args, get_pytest_rootdir):
+    with self._pants_pytest_plugin_args(sources_map) as plugin_args:
       with self._maybe_emit_coverage_data(workdirs,
                                           test_targets,
                                           pytest_binary.pex) as coverage_args:
-        yield (pytest_binary,
-               ['-p', pytest_binary.pytest_plugin_module] + plugin_args + coverage_args,
-               get_pytest_rootdir)
+        pytest_rootdir = get_buildroot()
```

Comment:
Is this still the right value in case of `--chroot`?

Reply:
Yes. There is code here that calculates the mapping between the chroot paths and the buildroot original paths:
https://github.com/pantsbuild/pants/blob/6f6060e0e9265e3a73b9a92aff9f9d4a6f3afb27/src/python/pants/backend/python/tasks/pytest_run.py#L476-L486

And maps back from junit.xml here:
https://github.com/pantsbuild/pants/blob/6f6060e0e9265e3a73b9a92aff9f9d4a6f3afb27/src/python/pants/backend/python/tasks/pytest_run.py#L519-L524